buildpack.go

```diff
@@ -95,6 +96,23 @@ type BuildpackDependency struct {
 	DeprecationDate time.Time `toml:"deprecation_date"`
 }
 
+// Equals compares the 2 structs if they are equal. This is very simiar to reflect.DeepEqual
+// except that properties that will not work (e.g. DeprecationDate) are ignored.
+func (b1 BuildpackDependency) Equals(b2 BuildpackDependency) bool {
+
+	b1.DeprecationDate = time.Time{}
+	b2.DeprecationDate = time.Time{}
+
+	if len(b1.CPEs) == 0 {
+		b1.CPEs = nil
+	}
+	if len(b2.CPEs) == 0 {
+		b2.CPEs = nil
+	}
+
+	return reflect.DeepEqual(b1, b2)
```

Comment:
```suggestion
	b1.DeprecationDate = b1.DeprecationDate.Truncate(time.Second).In(time.UTC)
	b2.DeprecationDate = b2.DeprecationDate.Truncate(time.Second).In(time.UTC)

	if len(b1.CPEs) == 0 {
		b1.CPEs = nil
	}
	if len(b2.CPEs) == 0 {
		b2.CPEs = nil
	}

	return reflect.DeepEqual(b1, b2)
```

I think if we truncate to the second & convert to UTC we can normalize the depreciation dates but still get a reasonable comparison. We should need to really even compare to the second, since it's just a *date*, but not sure if we want to go that far.

Thoughts?

Reply:
Like discussed in https://github.com/paketo-buildpacks/libpak/pull/233, I do not really see the point in comparing the metadata in the first place. And when it comes to `deprecationDate`, it never worked anyway. But that fixes the issue as well (I still think overly complicated, but it fixes it) and the cache misses are really annoying when downloading the same `JDK` over and over again.